packages/wrangler/src/config/validation.ts

```diff
@@ -2595,8 +2598,11 @@ const validateR2Binding: ValidatorFn = (diagnostics, field, value) => {
 		isValid = false;
 	}
 	if (
-		!isRequiredProperty(value, "bucket_name", "string") ||
-		(value as { bucket_name: string }).bucket_name.length === 0
+		getFlag("RESOURCES_PROVISION")
+			? !isOptionalProperty(value, "bucket_id", "string") ||
+				(value.bucket_id !== undefined && value.bucket_id.length === 0)
```

Comment:
I think this should be `bucket_name`?

Reply:
LOL. Good catch!

Reply:
https://github.com/cloudflare/workers-sdk/pull/7377/commits/73be97597160a7e01668fefe4d5a55a965ca2fad